### PR TITLE
capture (ticdc): fix http status panic

### LINKED ISSUE #5639: cdc panic in cross region replication
### What did you do?

1. create a primary cluster on dbaas
2. create a secondary cluster on dbaas
3. load some data to primary cluster(use betting workload)

### What did you expect to see?

cdc work normal

### What did you see instead?

there is a cdc panic in cdc logs
TyTSyM6Lws

### Versions of the cluster

5.4.1

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #5639 

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Fix a bug that may cause cdc server panic if it received a http request before cdc server fully started.
```

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

File: cdc/capture/capture.go [hunk 1]
```diff
@@ -316,7 +316,12 @@ func (c *Capture) run(stdCtx context.Context) error {
 func (c *Capture) Info() model.CaptureInfo {
 	c.captureMu.Lock()
 	defer c.captureMu.Unlock()
-	return *c.info
+	res := model.CaptureInfo{}
```
[the remark this answers is not in the document]
Comment (asddongmen) [reply cdc/capture/capture.go:319]: 1. It seems a problem, I will consider it more.
2. Good advice! thanks.

**Comment by sdojjy** [comment]
it's better to add gin recover middleware to fix all API panics

**Review by sdojjy — APPROVED** [review]

**Pull request merged**